swift/extractor/SwiftExtractor.cpp

```diff
@@ -211,3 +212,40 @@ void codeql::extractSwiftFiles(SwiftExtractorState& state, swift::CompilerInstan
     }
   }
 }
+
+static void cleanupPendingDeclarations(SwiftExtractorState& state) {
+  std::vector<const swift::Decl*> worklist;
+  std::copy(std::begin(state.pendingDeclarations), std::end(state.pendingDeclarations),
+            std::back_inserter(worklist));
+
+  for (auto decl : worklist) {
+    if (state.emittedDeclarations.count(decl)) {
+      state.pendingDeclarations.erase(decl);
+    }
+  }
```

Comment:
you don't need to create a vector from a set to filter out its elements, if you iterate using iterators, by using the fact that `erase` returns the iterator following the removed element:
```suggestion
  for (auto it = std::begin(state.pendingDeclarations); it != std::end(state.pendingDeclarations);) {
    if (state.emittedDeclarations.count(*it)) {
      it = state.pendingDeclarations.erase(it);
    } else {
      ++it;
    }
  }
```

Reply:
I think the worklist approach is more readable/less error prone.